### ISSUE TITLE
Crash/NPE @ onRequestPermissionsResult(PermissionManager.java:133)

### ISSUE THREAD

**Issue by PieterAelse** [Baseflow/flutter-permission-handler#665]
## 🐛 Bug Report

**See attached video:** 
Can be viewed/downloaded here: https://photos.app.goo.gl/Cp8Yx73cnmDSXakeA 

When changing the location permission from "Ask every time" or "While using the app" to "Always" **and** changing the new Android 12 behavior for the location permission from "Precise" to "Approximate" (in this order of steps) the plugin crashes the app:

> Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'boolean java.util.Map.containsKey(java.lang.Object)' on a null object reference
> at com.baseflow.permissionhandler.PermissionManager.onRequestPermissionsResult(PermissionManager.java:133)
> at io.flutter.embedding.engine.FlutterEngineConnectionRegistry$FlutterEngineActivityPluginBinding.onRequestPermissionsResult(FlutterEngineConnectionRegistry.java:713)

### Expected behavior
App/plugin doesn't crash.

### Reproduction steps

**Issue starts at step 13**, other steps are needed to get there.

1. Download/install the Philips Hue app on a device running Android 12 beta (in my case SPB5.210812.002 to be precise).
2. Setup the app (connect to a .bridge, lights etc).
3. Go to the tab "Automations".
4. Tap on the blue "Create automation" button, or the blue "+" button, _this will start the creation flow._
5. Select "Coming Home" (or "Leaving Home) automation.
6. Tap on the blue "Add this device" button, _this will start a setup flow (permissions etc)._
7. Tap "Next" until you need to "Enable out of home" control.
8. Enable "out of home" control by logging in, _proceeding to the next step._
9. Tap on "Allow access" to get the first permission pop-up.
10. Select "Precise" and either "While using the app" or "Only this time" _(the latter choice doesn't matter much). The next step will show a GoogleMaps screen with your current location._
11. Tap on "Save this location". _The next step will show info about battery optimization._
12. Tap on "Turn off" and "Allow" on the system dialog that pops up. _Next step will be "complaining" that the permission needs to be the "Always" type, not "While using the app" or "Only this time"._
13. Tap on "Allow access" and in the system dialog tap on "Allow in settings". _The permission settings screen will be shown._
14. Tap on "Allow all the time" **and** flip the toggle of "Use precise location" to **off**, in that order! (doing it the other way around didn't result in crash for me).
15. Leave the screen with e.g. the Android hardware back button/swipe or the back button on the top left.
16. ==> the app crashes on a NPE in `permission_handler` code (but maybe the issue is in Android 12 beta??)

Video's of this crash can be viewed/downloaded here: https://photos.app.goo.gl/Cp8Yx73cnmDSXakeA 

### Configuration

**Version:** `8.1.2` and reproduced on `8.1.4+2` as well.

**Platform:** 
- [ ] :iphone: iOS
- [X] :robot: Android 12 beta

### StackTrace

<details>
  <summary>Logs</summary>
  
```console
    --------- beginning of crash
2021-09-09 15:47:37.825 5501-5501/com.philips.lighting.hue2.daily E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.philips.lighting.hue2.daily, PID: 5501
    java.lang.RuntimeException: Unable to resume activity {com.philips.lighting.hue2.daily/com.philips.lighting.hue2.ContentActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=24, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {com.philips.lighting.hue2.daily/com.philips.lighting.hue2.ContentActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'boolean java.util.Map.containsKey(java.lang.Object)' on a null object reference
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4761)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4794)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:54)
        at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2214)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=24, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {com.philips.lighting.hue2.daily/com.philips.lighting.hue2.ContentActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'boolean java.util.Map.containsKey(java.lang.Object)' on a null object reference
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5304)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4748)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4794) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:54) 
        at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2214) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7842) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'boolean java.util.Map.containsKey(java.lang.Object)' on a null object reference
        at com.baseflow.permissionhandler.PermissionManager.onRequestPermissionsResult(PermissionManager.java:133)
        at io.flutter.embedding.engine.FlutterEngineConnectionRegistry$FlutterEngineActivityPluginBinding.onRequestPermissionsResult(FlutterEngineConnectionRegistry.java:712)
        at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.onRequestPermissionsResult(FlutterEngineConnectionRegistry.java:412)
        at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onRequestPermissionsResult(FlutterActivityAndFragmentDelegate.java:627)
        at io.flutter.embedding.android.FlutterActivity.onRequestPermissionsResult(FlutterActivity.java:648)
        at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:8529)
        at android.app.Activity.dispatchActivityResult(Activity.java:8386)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5297)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4748) 
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4794) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:54) 
        at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2214) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7842) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
```

</details>

**Comment by PieterAelse** [comment]
@mvanbeusekom Might be related to #659 I've just noticed

**Comment by florissmit1** [comment]
I started working on making a fix for this issue, see: #666. It will fix the `NullPointerException` when running your application in `release` mode. It's good to know, that the mentioned `use-case` is not solved when running in development mode. I guess this issue has nothing to do with the `NullPointerException` showing up when running in `release` mode.

**Comment by PieterAelse** [comment]
Verified that https://github.com/Baseflow/flutter-permission-handler/pull/666 fixes this issue, so closing it. Thanks!
